src/test/ui/nll/closure-requirements/propagate-approximated-ref.stderr

```diff
@@ -42,6 +42,12 @@ LL | fn supply<'a, 'b>(cell_a: Cell<&'a u32>, cell_b: Cell<&'b u32>) {
 ...
 LL |         demand_y(x, y, x.get())
    |         ^^^^^^^^^^^^^^^^^^^^^^^ argument requires that `'a` must outlive `'b`
+   |
+   = help: Consider adding the following bound: `'a: 'b`
```

Comment:
I would like it if this was a suggestion in the future, but it is fine as is now.

Reply:
@estebank I originally tried to implement this way-back-when, but it is very hard to get the correct span for the suggestion because there are a lot of possible cases:
- both lifetimes are defined on the `fn` in `<...>`: this is the easy case
- one or more lifetimes is inferred: this means we need to suggest that the user explicitly declare it, which possibly means suggesting that they add generics (`<...>`) to the `fn` header.
- one or more of the lifetime is defined on an enclosing scope (e.g. an `impl` block)
- a mix of the above could also happen

Moreover, their doesn't appear to be existing functionality for getting spans for most of the above.

Reply:
That's fine. I know that @zackmdavis wrote a bunch of helper methods (as used in https://github.com/rust-lang/rust/pull/62453), but I would like to take some time to clean this up throughout the compiler at a later date. Relying on text instead of suggestions is always better than not displaying information we can currently get.